lib/fastly_nsq/message_queue/strategy.rb

```diff
@@ -1,12 +1,34 @@
-class Strategy
-  def self.for_queue
-    if MessageQueue::FALSY_VALUES.include?(ENV['FAKE_QUEUE'])
-      Nsq
-    elsif MessageQueue::TRUTHY_VALUES.include?(ENV['FAKE_QUEUE'])
-      FakeMessageQueue
-    else
-      message = "You must set ENV['FAKE_QUEUE'] to either true or false"
-      raise InvalidParameterError, message
-    end
+module MessageQueue::Strategy
+  module_function
+
+  def for_queue
+    real_queue || fake_queue || error
+  end
+
+  private_class_method
```

Comment:
It's kind of Ruby minutiae: `private` overrides `module_function`, they don't "synergize". If you want a private, class-level method (or module-level method, as this case would have it), `private_class_method` is the thing. Personally, I think there's [an argument to made](http://blog.codeclimate.com/blog/2012/11/14/why-ruby-class-methods-resist-refactoring/) that these class-(module-)level methods aren't great, but I think it's the best scenario here. (aka "the tradeoffs are worth it")